src/plans/vectorial_plan.jl

```diff
@@ -262,54 +296,517 @@ function VectorHessianFunction(
     )
 end
 
-@doc raw"""
-    get_value(M::AbstractManifold, vgf::AbstractVectorFunction, p[, i=:])
+_vgf_index_to_length(b::BitVector, n) = sum(b)
+_vgf_index_to_length(::Colon, n) = n
+_vgf_index_to_length(i::AbstractArray{<:Integer}, n) = length(i)
+_vgf_index_to_length(r::UnitRange{<:Integer}, n) = length(r)
 
-Evaluate the vector function [`VectorGradientFunction`](@ref) `vgf` at `p`.
-The `range` can be used to specify a potential range, but is currently only present for consistency.
+#
+#
+# ---- Hessian
+@doc raw"""
+    get_hessian(M::AbstractManifold, vgf::VectorHessianFunction, p, X, i)
+    get_hessian(M::AbstractManifold, vgf::VectorHessianFunction, p, X, i, range)
+    get_hessian!(M::AbstractManifold, X, vgf::VectorHessianFunction, p, X, i)
+    get_hessian!(M::AbstractManifold, X, vgf::VectorHessianFunction, p, X, i, range)
 
-The `i` can be a linear index, you can provide
+Evaluate the Hessians of the vector function `vgf` on the manifold `M` at `p` in direction `X`
+and the values given in `range`, specifying the representation of the gradients.
 
+Since `i` is assumed to be a linear index, you can provide
 * a single integer
 * a `UnitRange` to specify a range to be returned like `1:3`
 * a `BitVector` specifying a selection
 * a `AbstractVector{<:Integer}` to specify indices
-* `:` to return the vector of all gradients, which is also the default
+* `:` to return the vector of all gradients
```

Comment:
Since it's `get_hessian`, it's probably not gradients?

Reply:
Yes, that was a copy paste error before that I also did not see when unifying the doc strings here. It should read “vector of all Hessian evaluations” (it still returns a vector of tangent vectors).